python/deltalake/table.py

```diff
@@ -473,15 +478,48 @@ def update(
         >>> from deltalake import DeltaTable
         >>> dt = DeltaTable("tmp")
         >>> dt.update(updates = {
-        ...             "deleted": True,
+        ...             "deleted": 'True',
         ...             "id": "concat(id, '_old')"
         ...             }
         ...         )
 
+        Update some row values with python object. This is equivalent to
+        ``UPDATE table SET price = 150.10 WHERE id = '5'``
+        >>> from deltalake import DeltaTable
+        >>> dt = DeltaTable("tmp")
+        >>> dt.update(predicate="id = '5'",
+        ...           updates = {
+        ...             "price": 150.10,
+        ...             }
+        ...         )
         """
-
+        if updates is None and new_values is not None:
+            updates = {}
+            for key, value in new_values.items():
+                if isinstance(value, (int, float, bool, list)):
```

Comment:
Are we sure this works for `list`? I believe DataFusion uses PostgreSQL syntax, which uses `{}` and not `[]` delimiters for arrays: https://www.postgresql.org/docs/current/arrays.html
Regardless, we should make sure to have a unit test for all of these types.

Reply:
Huh I guess you do have a test. I'm somewhat surprised it works, but I'll take it.

Reply:
There is also another weird quirk, I described it in the description of the PR.